arbor/memory/allocator.hpp

```diff
@@ -184,7 +184,7 @@ class allocator : public Policy {
         return &r;
     }
 
-    pointer allocate(size_type cnt, typename std::allocator<void>::const_pointer = 0) {
+    pointer allocate(size_type cnt, const_pointer = 0) {
```

Comment:
Now that we live in C++17 land, this could be simplified to
```
pointer allocate(size_type cnt)
```
https://en.cppreference.com/w/cpp/memory/allocator/allocate

Reply:
Good point.